### PR TITLE
Mac build scripts: consume git installer pkg directly

### PR DESCRIPTION
**Pull request by jamill**
Update VFS4G build scripts to consume the macOS pkg installer package directly

Previously, the GitForMac.GVFS.Installer NuGet package contained only
the git for mac installer as a .dmg file. This .dmg package contained
the Git for Mac installer .pkg.

The VFS4G installer takes a .pkg installer as input to the combined
VFS4G / Git for Mac installer. As the NuGet package only contained the
.dmg, the script to build the VFS4G combined installer had to mount
the .dmg and extract the .pkg file.

The GitForMac.GVFS.Installer package has been updated to include both
a .dmg and .pkg file, so the installer script can now consume the
installer .pkg directory, instead of mounting and extracting it from
the .dmg file.

This change also relaxes the regex pattern to correctly handle git versions with multiple hyphens. Previously, this would fail to capture the full version string for versions such as:

1.2.3-DRAFT-pr

### CODE REVIEW

**Comment by wilbaker** [comment]
Could you update the description with more details on why these changes were needed?

File: Scripts/Mac/GetGitVersionNumber.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 . "$(dirname ${BASH_SOURCE[0]})/InitializeEnvironment.sh"
 
 GVFSPROPS=$VFS_SRCDIR/GVFS/GVFS.Build/GVFS.props
-GITVERSION="$(cat $GVFSPROPS | grep GitPackageVersion | grep -Eo '[0-9.]+(-\w+)?')"
-echo $GITVERSION
\ No newline at end of file
+GITVERSION="$(cat $GVFSPROPS | grep GitPackageVersion | grep -Eo '[0-9.]+(-\w+)*')"
```
Comment (wilbaker) [inline Scripts/Mac/GetGitVersionNumber.sh]: Why was this change needed?

File: GVFS/GVFS.Installer.Mac/CreateMacInstaller.sh [hunk 2]
```diff
@@ -179,49 +179,34 @@ function CreateVFSForGitDistribution()
 function CreateMetaDistribution()
 {
     GITVERSION="$($VFS_SCRIPTDIR/GetGitVersionNumber.sh)"
-    GITDMGPATH="$(find $VFS_PACKAGESDIR/gitformac.gvfs.installer/$GITVERSION -type f -name *.dmg)" || exit 1
-    GITDMGNAME="${GITDMGPATH##*/}"
-    GITINSTALLERNAME="${GITDMGNAME%.dmg}"
-    GITVERSIONSTRING=`echo $GITINSTALLERNAME | cut -d"-" -f2`
-    GITPKGNAME="$GITINSTALLERNAME.pkg"
-    
+    GITINSTALLERPKGPATH="$(find $VFS_PACKAGESDIR/gitformac.gvfs.installer/$GITVERSION -type f -name *.pkg)" || exit 1
```
Comment (wilbaker) [inline GVFS/GVFS.Installer.Mac/CreateMacInstaller.sh]: I've not been following much of the installer discussions, what are the pros of a .pkg (over a .dmg)?

**Comment by jamill** [comment]
> Could you update the description with more details on why these changes were needed?

Thanks - the commits themselves contained descriptive descriptions, but I didn't update the PR description. I updated the PR description to be more meaningful.

File: Scripts/Mac/GetGitVersionNumber.sh (hunk 1 above)
Comment (wilbaker) [reply Scripts/Mac/GetGitVersionNumber.sh]: See the answer in the description now, thanks!

**Comment by wilbaker** [comment]
> The GitForMac.GVFS.Installer package has been updated to include both
a .dmg and .pkg file, so the installer script can now consume the
installer .pkg directory, instead of mounting and extracting it from
the .dmg file.

Is there something/someone who is consuming the .dmgs?  I am not sure if making a .dmg available is a standard Apple thing.

**Comment by jamill** [comment]
> Is there something/someone who is consuming the .dmgs? I am not sure if making a .dmg available is a standard Apple thing.

Great question. After we transition this build off of that requirement, maybe we can stop publishing the .dmg.
